etcdmgmt/etcd-mgmt.go

```diff
@@ -160,3 +151,98 @@ func isETCDStartNeeded() (bool, int) {
 	}
 	return start, pid
 }
+
+// Check whether etcd environment variable present or not
+// If it present then start etcd without --initial-cluster flag
+// other wise start etcd normally.
+func ETCDStartInit() (*os.Process, error) {
+
+	file, err := os.Open(etcdConfFile)
+	if err != nil {
+		switch {
+		case os.IsNotExist(err):
+			log.Info("Starting/Restarting etcd for a initial node")
+			return StartStandAloneETCD()
+		default:
+			log.WithFields(log.Fields{
+				"error": err,
+				"path":  etcdPidFile,
+			}).Fatal("Failed to read from file")
+		}
+	} else {
+		defer file.Close()
+
+		// Restoring etcd environment variable and starting etcd daemon
+		scanner := bufio.NewScanner(file)
+		scanner.Split(bufio.ScanLines)
+		for scanner.Scan() {
+			linestr := scanner.Text()
+			etcdenv := strings.Split(linestr, "=")
+
+			etcdEnvKey := etcdenv[0]
+			etcdEnvData := etcdenv[1]
+			envlen := len(etcdenv)
+			for i := 2; i < envlen; i++ {
+				etcdEnvData = etcdEnvData + "=" + etcdenv[i]
+			}
+
+			// setting etcd environment variable
+			os.Setenv(etcdEnvKey, etcdEnvData)
+		}
+
+		args := []string{"-listen-client-urls", listenClientUrls,
+			"-advertise-client-urls", advClientUrls,
+			"-listen-peer-urls", listenPeerUrls,
+			"-initial-advertise-peer-urls", initialAdvPeerUrls}
+
+		log.Info("Sstarting etcd daemon")
+		return StartETCD(args)
+	}
+	return nil, err
+}
+
+// Starting default etcd by considering single server node
```

Comment:
I didn't know about this format and saw lot of place in current existing code where nothing there as a comment in current existing code. Any way i will do this change.